### PR TITLE
expose as ES6 in addition to UMD

### PR DESCRIPTION
**Pull request by CarstenLeue**
Added a build step to build ES6 output in addition to UMD style output. The effect is that webpack based tools can now treeshake the component.

##### Checklist

- [x] `npm test` passes (tip: `npm run lint-fix` can correct most style issues)
- [ ] tests are included
- [ ] documentation is changed or added

### CODE REVIEW

**Comment by CarstenLeue** [comment]
Example of the effect of the es6 compile for an application that only contains this snippet (https://github.com/CarstenLeue/node-sdk-core-tree-shaking):

```typescript
import { getAuthenticatorFromEnvironment } from 'ibm-cloud-sdk-core';

const auth = getAuthenticatorFromEnvironment('SECRETS_MANAGER_API');
console.log('auth', auth);
```

with tree shaking:
image

without tree shaking (i.e. before this PR):
image

Notice that the tree-shaken version does no longer contain the expect, jest and babel dependencies.

Both versions are still pretty large, but this is a consequence of using the `mime-types` and `tough-cookie` dependencies. If you are interested in providing a compact layer, you might want to revisit these dependencies.

**Comment by CarstenLeue** [comment]
@dpopp07 when building the `ibm-cloud-sdk-core` via `npm run all` I notice that the `dist` folder is populated but there is no `package.json` inside that dist folder. Maybe this gets added via the final publish step. 
It's important that in the `package.json` that finally gets published you find the `module` and `sideEffects` entry (https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free)

**Comment by dpopp07** [comment]
@CarstenLeue I would support being more explicit in adding the `package.json` to the `dist` folder if you want. You could add a copy step to the build step, we've done this in other packages where we needed to be more explicit, e.g.

`"build": "tsc && cp package.json dist/",`

**Comment by CarstenLeue** [comment]
@dpopp07 sure. I have added a step to copy the `package.json` into the dist folder. I am not using `cp` since that does not work cross platform. Also I am removing the sections from the copied `package.json` that should not be published anyway.

**Review by dpopp07 — COMMENTED** [review]
This looks great! Just have one question to address before approving

**Comment by dpopp07** [comment]
@CarstenLeue is there anything we need to do manually to support the new documentation related additions in this PR?

File: etc/ibm-cloud-sdk-core.api.md [hunk 1]
```diff
@@ -0,0 +1,458 @@
+## API Report File for "ibm-cloud-sdk-core"
```
Comment (padamstx) [inline etc/ibm-cloud-sdk-core.api.md:1]: Perhaps this one also should not be tracked in git???

File: es/tsdoc-metadata.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// This file is read by tools that parse documentation comments conforming to the TSDoc standard.
+// It should be published with your NPM package.  It should not be tracked by Git.
```
[the remark this answers is not in the document]
Comment (CarstenLeue) [reply es/tsdoc-metadata.json:2]: let me remove it

**Comment by CarstenLeue** [comment]
> @CarstenLeue is there anything we need to do manually to support the new documentation related additions in this PR?

@dpopp07 maybe not manually but there should be a step in the automation that publishes the npm package. This step should also write the generated doc files to the `gh-pages` branch, e.g. like so https://docs.travis-ci.com/user/deployment/pages/

But I am lacking information how the actual build integration is working

File: etc/ibm-cloud-sdk-core.api.md (hunk 1 above)
Comment (CarstenLeue) [reply etc/ibm-cloud-sdk-core.api.md:1]: This file is supposed to be tracked https://api-extractor.com/pages/overview/demo_api_report/
It contains the description of the public API surface. Tracking it allows you to detect intended or unintended changes to the public API.

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**